ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/IMEIMaskingProvider.java

```diff
@@ -10,36 +10,53 @@
 import com.ibm.whc.deid.util.IMEIManager;
 import com.ibm.whc.deid.util.RandomGenerators;
 
+/**
+ * Privacy provider to mask International Mobile Equipment Identity (IMEI) device identifiers.
+ */
 public class IMEIMaskingProvider extends AbstractMaskingProvider {
-  /** */
+
   private static final long serialVersionUID = -5189348513611132622L;
 
-  private static final IMEIIdentifier identifier = new IMEIIdentifier();
+  private static final IMEIIdentifier imeiIdentifier = new IMEIIdentifier();
   private static final IMEIManager imeiManager = new IMEIManager();
+
   private final boolean preserveTAC;
+  private final int unspecifiedValueHandling;
+  private final String unspecifiedValueReturnMessage;
 
   /**
-   * Instantiates a new Imei masking provider.
+   * Instantiates a new IMEI masking provider.
    *
-   * @param config an IMEIMaskingProviderConfig instance
+   * @param config privacy provider configuration
    */
   public IMEIMaskingProvider(IMEIMaskingProviderConfig config) {
     this.preserveTAC = config.getPreserveTAC();
+    unspecifiedValueHandling = config.getUnspecifiedValueHandling();
+    unspecifiedValueReturnMessage = config.getUnspecifiedValueReturnMessage();
   }
 
   @Override
   public String mask(String identifier) {
-    String tac;
-    if (!IMEIMaskingProvider.identifier.isOfThisType(identifier)) {
+
+    if (identifier == null) {
+      debugFaultyInput("null");
       return null;
```

Comment:
I think we shouldn't return null here but instead follow what the "unspecifiedValueHandling" value says to do in this case.

Reply:
This is a safety check.  Currently, if the input value is null, no call is made to the providers.  The code that determines what should be masked doesn't consider this a leaf.  Most of the providers, but not all, currently have a check like this.